Framework/CurveFitting/src/Functions/InelasticDiffRotDiscreteCircle.cpp

```diff
@@ -150,11 +150,11 @@ void InelasticDiffRotDiscreteCircle::setWorkspace(
       break;
     }
 
-    const auto &det = spectrumInfo.detector(idx);
+    const auto detID = spectrumInfo.detector(idx).getID();
```

Comment:
I have just pushed a change to try and do this. You cannot use `idx` as the index, as the `SpectrumInfo` and `DetectorInfo` indexes refer to different things, but you can do this with the `SpectrumDefinition`. Could you have a careful check of what I have done - I think taking the first detector ID is correct for what would have happened previously for groups, but I am not 100% sure.